### PR TITLE
Fix false calculation of w (omega) when resolveInFrame == frame_resolve

### PR DESCRIPTION
**Pull request by tobolar**
Back-port #4148 from master to maint/4.0.x (as suggested in https://github.com/modelica/ModelicaStandardLibrary/pull/4148#issuecomment-1793827786)

### CODE REVIEW

**Review by HansOlsson — APPROVED** [review]
Looks good.

File: ModelicaTest/Resources/Reference/ModelicaTest/MultiBody/Sensors/AbsoluteSensor/comparisonSignals.txt
Comment (beutlich) [inline ModelicaTest/Resources/Reference/ModelicaTest/MultiBody/Sensors/AbsoluteSensor/comparisonSignals.txt:1]: These changes do not belong to this PR.

Comment (tobolar) [reply ModelicaTest/Resources/Reference/ModelicaTest/MultiBody/Sensors/AbsoluteSensor/comparisonSignals.txt:1]: You are right. Rebased now, without that file.

**Review by beutlich — APPROVED** [review]
